### PR TITLE
Renaming and javadocs around SlotAllocation

### PR DESCRIPTION
**Pull request by fickludd**
PipelineInformation is now SlotConfiguration

### CODE REVIEW

**Comment by henriknyman** [comment]
Couldn't we do this together with an introduce `PhysicalPlan` PR to minimize the merge pain?

**Comment by fickludd** [comment]
@henriknyman I suggest to introduce `PhysicalPlan` in https://github.com/neo4j/neo4j/pull/10460, which builds on this PR. Is that an OK solution?

File: enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotConfiguration.scala [hunk 1]
```diff
@@ -159,25 +68,24 @@ class PipelineInformation(private val slots: mutable.Map[String, Slot],
...
-  def add(key: String, slotInformation: Slot): Unit = slotInformation match {
+  def add(key: String, slot: Slot): Unit = slot match {
     case LongSlot(_, nullable, typ) => newLong(key, nullable, typ)
     case RefSlot(_, nullable, typ) => newReference(key, nullable, typ)
   }
 
-  def breakPipelineAndClone(): PipelineInformation = {
-    val newPipeline = new PipelineInformation(this.slots.clone(), numberOfLongs, numberOfReferences)
+  def copy(): SlotConfiguration = {
+    val newPipeline = new SlotConfiguration(this.slots.clone(), numberOfLongs, numberOfReferences)
```
Comment (pontusmelke) [inline enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotConfiguration.scala:180]: Maybe would be nicer to have alias and slotAlias part of constructor,
something like

```
class SlotConfiguration(private val slots: mutable.Map[String, Slot],
                          val initialNumberOfLongs: Int,
                          val initialNumberOfReferences: Int,
                          aliases: mutable.Set[String] = mutable.Set(),
                          slotAliases: SlotMap = new SlotMap
                ) 
```
where
```
type SlotMap =  mutable.HashMap[Slot, mutable.Set[String]] with mutable.MultiMap[Slot, String]
```

**Comment by systay** [comment]
This has been annoying me for some time (I realise it's mostly my fault). Thanks for taking it on!

File: enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotConfiguration.scala (hunk 1 above)
Comment (fickludd) [reply enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotConfiguration.scala:180]: I find this change attractive, but wasn't very happy with the result when I tried to do it. Maybe we can change that in a next step? I get ugliness when cloning the `SlotMap` type.

**Comment by fickludd** [comment]
@systay it's been annoying me too! I just had to fix it before continuing.

**Pull request merged**